specification/compute/resource-manager/Microsoft.Compute/preview/2018-06-01/gallery.json

```diff
@@ -0,0 +1,1120 @@
+    },
+    "GalleryOSDiskImage": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/GalleryDiskImage"
+        }
+      ],
+      "description": "This is the OS disk image."
+    },
+    "GalleryDataDiskImage": {
+      "properties": {
+        "lun": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Specifies the logical unit number of the data disk. This value is used to identify data disks within the VM and therefore must be unique for each data disk attached to a VM."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/GalleryDiskImage"
+        }
+      ],
+      "description": "This is the data disk image."
+    },
+    "GalleryDiskImage": {
+      "properties": {
+        "sizedInGB": {
+          "type": "integer",
+          "format": "int32",
+          "description": "It indicates the size of the VHD to create."
+        },
+        "hostCaching":{
+          "type": "string",
+          "description": "The host caching of the disk. Valid values are 'None', 'ReadOnly', and 'ReadWrite'",
+          "enum": [
+            "None",
+            "ReadOnly",
+            "ReadWrite"
+          ],
+          "x-ms-enum": {
+            "name": "HostCaching",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "This is the disk image base class."
+    },
+    "ReplicationStatus": {
+      "properties": {
+        "aggregatedState": {
+          "type": "string",
+          "description": "This is the aggregated replication status based on the regional replication status.",
+          "enum": [
+            "Unknown",
+            "InProgress",
+            "Completed",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "AggregatedReplicationState",
+            "modelAsString": false
+          }
+        },
+        "summary": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RegionalReplicationStatus"
+          },
+          "description": "This is a summary of replication status for each region."
+        }
+      },
+      "description": "This is the replication status of the gallery image version."
+    },
+    "RegionalReplicationStatus": {
+      "properties": {
+        "region": {
+          "type": "string",
+          "description": "The region where the gallery image version is published to."
+        },
+        "state": {
+          "type": "string",
+          "description": "This is the regional replication state.",
+          "enum": [
+            "Unknown",
+            "Replicating",
+            "Completed",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ReplicationState",
+            "modelAsString": false
+          }
+        },
+        "details": {
+          "type": "string",
+          "description": "The details of the replication status."
+        },
+        "progress": {
```

Comment:
why is this not a part of provisioningState itself? when a PUT on a imageVersion happens, the PUT will not be complete till the replication in all regions have completed, correct? If yes, then provisioningState should be used to reflect replication status. If the PUT can complete but replication can still be in progress, then is it usable? I guess it depends on the region where its available etc. Does it make sense in this scenario to tie replication with provisioning to provide a better customer experience. As all the clients know how to poll using the provisioningState and can reflect the accurate status back to customer.

Reply:
This is a long running operation with a timeout. The operation may fail after the timeout even if the replication is still in progress. If the PUT completes (fails) with replication still in progress, customers can do a GET to check the regional replication status (state, detailed error message, and progress).

Reply:
This will be misleading to the customers. PUT has failed but operation (replication) is still in progress.

Reply:
Since we are replicating big storage blobs, the replication job may got stuck, which may happen occasionally. In that case, the replication time may exceed our timeout and we regard that as failure.